### PR TITLE
test: install `include_subdirs qualified` library and a multi-segment name

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

File: test/blackbox-tests/test-cases/install/dune-package-source-path.t [hunk 1]
```diff
@@ -43,3 +43,61 @@ Test paths on public libraries with `.` are correct
...
+
+
+  $ dune build b.install --root b
+  Entering directory 'b'
+  Leaving directory 'b'
+
+  $ cat b/_build/default/b.install
+  lib: [
+    "_build/install/default/lib/b/META"
+    "_build/install/default/lib/b/child/bar.ml" {"child/bar.ml"}
```
Comment (anmonteiro) [inline test/blackbox-tests/test-cases/install/dune-package-source-path.t:73]: this should be `lib/b/sub/child/bar.ml`

File: test/blackbox-tests/test-cases/install/dune-package-source-path.t [hunk 2]
```diff
@@ -89,6 +87,8 @@ Test paths on public libraries with `.` are correct
     "_build/install/default/lib/b/sub/b__Foo.cmi" {"sub/b__Foo.cmi"}
     "_build/install/default/lib/b/sub/b__Foo.cmt" {"sub/b__Foo.cmt"}
     "_build/install/default/lib/b/sub/b__Foo.cmx" {"sub/b__Foo.cmx"}
+    "_build/install/default/lib/b/sub/child/bar.ml" {"sub/child/bar.ml"}
+    "_build/install/default/lib/b/sub/child/child.ml" {"sub/child/child.ml"}
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/install/dune-package-source-path.t:91]: So the values in the dune-package file are correct?

Comment (anmonteiro) [reply test/blackbox-tests/test-cases/install/dune-package-source-path.t:91]: I believe they are. Check the last line of this test.

**Comment by rgrinberg** [comment]
Thanks. I simplified the fix somewhat.

**Pull request merged**